### PR TITLE
Migration v8 experimental

### PR DESCRIPTION
**Pull request by ben-kaufman**

### CODE REVIEW

File: src/mappings/JoinAndQuit/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+type JoinAndQuitProposal @entity {
+	id: ID!
+	dao: DAO!
+	proposedMember: Bytes!
+	funder: Bytes!
+	funding: BigInt!
+	executed: Boolean!
```
Comment (orenyodfat) [inline src/mappings/JoinAndQuit/schema.graphql]: might need addition filed to say if the dao accept or reject the proposal.

File: src/mappings/JoinAndQuit/mapping.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import {
+  JoinInProposal,
+  ProposalExecuted,
+  RedeemReputation,
+} from '../../types/JoinAndQuit/JoinAndQuit';
+
```
Comment (orenyodfat) [inline src/mappings/JoinAndQuit/mapping.ts]: where is rageQuit is handled ?

File: src/mappings/JoinAndQuit/datasource.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+entities:
+  - JoinAndQuitProposal
+eventHandlers:
+  - event: ProposalExecuted(indexed address,indexed bytes32,int256)
+    handler: handleProposalExecuted
+  - event: JoinInProposal(indexed address,indexed bytes32,string,address,uint256)
+    handler: handleNewJoinAndQuitProposal
+  - event: FundedBeforeDeadline(indexed address)
+    handler: handleFundedBeforeDeadline
+  - event: Donation(indexed address,indexed uint256)
```
Comment (orenyodfat) [inline src/mappings/JoinAndQuit/datasource.yaml]: There is no Donation event .

File: src/mappings/JoinAndQuit/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+	reputationMinted: BigInt!
```
[the remark this answers is not in the document]
Comment (ben-kaufman) [reply src/mappings/JoinAndQuit/schema.graphql]: yes

File: src/mappings/JoinAndQuit/schema.graphql (hunk 1 above)
Comment (ben-kaufman) [reply src/mappings/JoinAndQuit/schema.graphql]: We have that in the Proposal entity. We could also remove the executed I guess.

File: src/mappings/JoinAndQuit/mapping.ts (hunk 2 above)
Comment (ben-kaufman) [reply src/mappings/JoinAndQuit/mapping.ts]: What should the subgraph do in a ragequit?

File: src/mappings/JoinAndQuit/datasource.yaml (hunk 3 above)
Comment (ben-kaufman) [reply src/mappings/JoinAndQuit/datasource.yaml]: Removed

File: src/mappings/JoinAndQuit/mapping.ts (hunk 2 above)
Comment (orenyodfat) [reply src/mappings/JoinAndQuit/mapping.ts]: index the event of that and for each funder decreased it funding.
does a ragequit member should get out of the dao is a good point.

File: src/mappings/FundingRequest/schema.graphql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+type FundingRequestProposal @entity {
+	id: ID!
```
Comment (orenyodfat) [inline src/mappings/FundingRequest/schema.graphql]: should it include also if its token or native coin ?
if token , so its address ?

Comment (ben-kaufman) [reply src/mappings/FundingRequest/schema.graphql]: It already includes it in the scheme parameters. No need to cache this per proposal as it is the same for all proposals of the scheme

File: src/mappings/JoinAndQuit/mapping.ts (hunk 2 above)
Comment (ben-kaufman) [reply src/mappings/JoinAndQuit/mapping.ts]: I will add that when you release the Arc v9 with the fix of the ragequit

Comment (orenyodfat) [reply src/mappings/JoinAndQuit/mapping.ts]: do you mean arc V11 ?

**Pull request merged**